commit 24514b9970f50224fd7df8a6b055030ee67ab9be (refs/changes/40/2740/2)
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2017-10-10T09:04:38-07:00 (2 years ago)
    
    DOCKER-1115 uncaught exception when docker log driver does not specify the type
    Reviewed by: Josh Wilsdon <josh@wilsdon.ca>
    Approved by: Josh Wilsdon <josh@wilsdon.ca>

diff --git a/lib/backends/sdc/containers.js b/lib/backends/sdc/containers.js
index d3e52c0..11938cd 100644
--- a/lib/backends/sdc/containers.js
+++ b/lib/backends/sdc/containers.js
@@ -1685,7 +1685,7 @@ function buildVmPayload(opts, container, callback) {
 
     // This was already validated in lib/validate.js
     if (container.HostConfig && container.HostConfig.LogConfig) {
-        if (container.HostConfig.LogConfig.Type.length > 0) {
+        if (container.HostConfig.LogConfig.Type) {
             payload.internal_metadata['docker:logdriver']
                 = container.HostConfig.LogConfig.Type;
         } else {
diff --git a/package.json b/package.json
index 599351c..9bef42e 100644
--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "sdc-docker",
-  "version": "0.4.7",
+  "version": "0.4.8",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
